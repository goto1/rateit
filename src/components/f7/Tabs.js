import React from "react";
import PropTypes from "prop-types";

const Tabs = ({ className, children }) => {
  const classNames = `tabs ${className ? className : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

Tabs.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default Tabs;
