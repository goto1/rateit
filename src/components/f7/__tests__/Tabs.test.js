import React from "react";
import Tabs from "../Tabs";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("Tabs component", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <Tabs>
          <span>HelloWorld!</span>
        </Tabs>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with a custom styling class", () => {
    const wrapper = shallow(<Tabs className="custom">HelloWorld!</Tabs>);
    const actual = wrapper.props().className.includes("custom");
    const expected = true;
    expect(actual).toBe(expected);
  });
});
