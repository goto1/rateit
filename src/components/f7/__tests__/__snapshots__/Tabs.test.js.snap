// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component should render correctly 1`] = `


<div
  className="tabs"
>
  <span>
    HelloWorld!
  </span>
</div>
`;
